core/src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -606,6 +608,9 @@ public synchronized void updateMetaData(final IndexMetaData metadata) {
             if (refreshTask.getInterval().equals(indexSettings.getRefreshInterval()) == false) {
                 rescheduleRefreshTasks();
             }
+            if (trimTranslogTask.getInterval().equals(indexSettings.getTranslogRetentionCheckInterval()) == false) {
```

Comment:
Can you elaborate why you go down this route instead of using `IndexShard#afterWriteOperation()` and maybe also do it in `IndexShard#flush`? I'd love to rather remove tasks than adding them?! it's just way easier to reason about things if they are happening due to a user interaction form the outside